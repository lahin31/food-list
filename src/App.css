* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.container {
  height: 100%;
  width: 100%;
  background: #000000;
}

.cover_pic {
  height: 100%;
  position: absolute;
  width: 100%;
  top: 0;
  right: 0;
  object-fit: cover;
  object-position: top;
}