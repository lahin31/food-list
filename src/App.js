import React, { useState, useEffect } from 'react';
import { TimelineMax, Expo } from 'gsap';
import Sidebar from "./components/Sidebar";
import "./App.css";

function App() {
  const [foods, setFoods] = useState([]); 
  const [animation, setAnimation] = useState(false);

  useEffect(() => {
    fetch(`${process.env.PUBLIC_URL}/db.json`)
      .then(res => res.json())
      .then(res => setFoods(res.foods))
  }, []);

  const coverImgToggle = () => {
    if(!animation) {
      showAnimation();
    } else {
      hideAnimation();
    }
  }

  const showAnimation = () => {
    const t1 = new TimelineMax();
    t1.to('.cover_pic', 1, {
      width: "60%",
      ease: Expo.easeOut
    }).to('nav', 1, {
      height: "100%",
      ease: Expo.easeOut
    }, "-=0.6").fromTo('.nav-open', 1, {
      opacity: 0,
      x: 50,
      ease: Expo.easeOut
    }, {
      opacity: 1,
      x: 0, 
      onComplete: function() {
        document.querySelector('.nav-open').style.pointerEvents = "auto";
      }
    });
    t1.play();
    setAnimation(true)
  }

  const hideAnimation = () => {
    const t1 = new TimelineMax();
    t1.fromTo('.nav-open', .7, {
      opacity: 1,
    }, {
      opacity: 0,
      x: 0,
      ease: Expo.easeOut 
    }).to('nav', 1, {
      height: "20%",
      ease: Expo.easeOut
    }).to('.cover_pic', 1, {
      width: "100%",
      ease: Expo.easeOut,
      onComplete: function() {
        document.querySelector('.nav-open').style.pointerEvents = "none";
      }
    }, "-=0.5");

    t1.play();
    setAnimation(false);
  }
  return ( 
    <div className="container">
      <Sidebar 
        foods={foods} 
        coverImgToggle={coverImgToggle}
      />
      <img 
        className="cover_pic" 
        src={`${process.env.PUBLIC_URL}/images/berries-pizza.jpg`} 
        alt={`Wallpaper`}
        data-testid="wallpaper"
      />
    </div>
  );
}
 
export default App;