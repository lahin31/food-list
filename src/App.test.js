import React from 'react';
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'
import App from "./App";

afterEach(cleanup);

it("makes a snapshot and tests", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it("tests the className for wallpaper", () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId('wallpaper')).toHaveClass('cover_pic');
});