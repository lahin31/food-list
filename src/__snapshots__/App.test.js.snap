// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makes a snapshot and tests 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header>
      <nav>
        <div
          class="nav-closed"
        >
          <h3
            data-testid="title"
          >
            Food App
          </h3>
          <ul
            class="nav-links"
          >
            <li
              class="nav-button"
            >
              Foods
            </li>
          </ul>
        </div>
        <div
          class="nav-open"
        >
          <div
            class="nav-images"
          />
        </div>
      </nav>
    </header>
    <img
      alt="Wallpaper"
      class="cover_pic"
      data-testid="wallpaper"
      src="/images/berries-pizza.jpg"
    />
  </div>
</DocumentFragment>
`;
